### ISSUE TITLE
Add ability to select date in 12-hour AM/PM format

### ISSUE THREAD

**Issue by stevendesu** [YouCanBookMe/react-datetime#168]
In the `calculateState` function of `src/TimeView.js` it creates counters for the hours, minutes, and seconds:

```js
// Lines 17 - 25
if ( format.indexOf('H') !== -1 || format.indexOf('h') !== -1 ){
	counters.push('hours');
	if ( format.indexOf('m') !== -1 ){
		counters.push('minutes');
		if ( format.indexOf('s') !== -1 ){
			counters.push('seconds');
		}
	}
}
```

However it does not create an additional counter for "AM/PM" when the format contains " A"

Therefore upon clicking the DateTime input you see a properly AM/PM formatted time:

![am-pm-time](https://cloud.githubusercontent.com/assets/1384974/19397224/ee7ae3a6-9214-11e6-9869-09f93eaa4b6a.jpg)

However if you click to change it, it reverts to 24-hour:

![24-hour-time](https://cloud.githubusercontent.com/assets/1384974/19397242/0792ccbe-9215-11e6-8858-2b3a6006f06c.jpg)

Ideally if the format contains " A" (or even "A" without the preceding space) it should recognize this and the time selector should look more like the following (forgive my crappy MSPaint skills):

![better-input](https://cloud.githubusercontent.com/assets/1384974/19397315/53a4cb66-9215-11e6-9f9b-62d1dec4a95a.jpg)

**Comment by simeg** [comment]
Hi @stevendesu. I can't reproduce this. If you check out [the demo](http://codepen.io/simeg/pen/mEmQmP) it works as intended. Can you please provide a code example which is not working as intended? Thanks.

**Comment by stevendesu** [comment]
I'll try to pull together the minimum necessary code to repro. I'm experiencing the issue on a fairly large project within my company so it could take some time to figure out.

**Comment by stevendesu** [comment]
Doing some further research it looks like this issue was resolved in v2.3.2 -- we were still using v2.1.0

I'll update the library and if the issue is resolved I'll close this

**Comment by stevendesu** [comment]
Looks like updating it resolved the issue! Sorry for the distraction 😄 Closing this now
